components/registration/registration-controller.js

```diff
@@ -311,19 +311,30 @@ trackerCapture.controller('RegistrationController',
         }
     };
 
+    $scope.$on('changeOrgUnit', function (event, args) {
+        $scope.tei.orgUnit = args.orgUnit;
+        $scope.registerEntity("newOrgUnit");
+    });
+
     var performRegistration = function (destination) {
         if (destination === "DASHBOARD" || destination === "SELF") {
            $scope.model.savingRegistration = true;
         }
-        RegistrationService.registerOrUpdate($scope.tei, $scope.optionSets, $scope.attributesById).then(function (registrationResponse) {
-            var reg = registrationResponse.response ? registrationResponse.response : {};
+        RegistrationService.registerOrUpdate($scope.tei, $scope.optionSets, $scope.attributesById).then(function (regResponse) {
+            var reg = regResponse.response.responseType ==='ImportSummaries' ? regResponse.response.importSummaries[0] : regResponse.response.responseType === 'ImportSummary' ? regResponse.response : {};
             if (reg.status === 'SUCCESS') {
-                $scope.tei.trackedEntityInstance = reg.importSummaries[0].reference;
+                $scope.tei.trackedEntityInstance = reg.reference;
                 
                 if ($scope.registrationMode === 'PROFILE') {
                     reloadProfileWidget();
                     $rootScope.$broadcast('teiupdated', {});
                     $scope.model.savingRegistration = false;
+                    if(destination==='newOrgUnit'){

```

Comment:
Please add some spaces in between the operator and the operands - like we have in *most* of the other code :)